jax/_src/lax/lax.py

```diff
@@ -2175,21 +2175,43 @@ def _minmax_complex_lowering(x, y, *, lax_cmp_pick_x):
 mlir.register_lowering(shift_right_logical_p,
                        partial(_nary_lower_hlo, hlo.ShiftRightLogicalOp))
 
+def _opaque_comparison_hlo(direction, reduction_op, identity, ctx,
+                           avals_in, aval_out, x, y):
+  aval_x, aval_y = avals_in
+  base_aval_x = core.physical_aval(aval_x)
+  base_aval_y = core.physical_aval(aval_y)
+  base_aval_out = core.ShapedArray(base_aval_x.shape, aval_out.dtype)
+  reduce_axes = tuple(range(aval_out.ndim, base_aval_out.ndim))
+  res, = mlir.delegate_lowering(
+      ctx, partial(_compare_lower_hlo, direction),
+      x, y, avals_in=[base_aval_x, base_aval_y], avals_out=[base_aval_out])
+  return mlir.delegate_lowering(
+      ctx, partial(_unary_reduce_lower, reduction_op, identity,
+                   axes=reduce_axes),
+      res, avals_in=[base_aval_out], avals_out=[aval_out])
+
+_opaque_eq_hlo = partial(
+    _opaque_comparison_hlo, 'EQ', hlo.AndOp, _get_bitwise_and_identity)
+_opaque_ne_hlo = partial(
+    _opaque_comparison_hlo, 'NE', hlo.OrOp, _get_bitwise_or_identity)
+
+def _compare_lower_hlo_opaque(direction: str, ctx, avals_in, aval_out, x, y):
+  broadcast_avals_in = tuple(
+      core.ShapedArray(aval_out.shape, aval.dtype) for aval in avals_in)
+  if direction == 'EQ':
+    return _opaque_eq_hlo(ctx, broadcast_avals_in, aval_out, x, y)
+  elif direction == 'NE':
+    return _opaque_ne_hlo(ctx, broadcast_avals_in, aval_out, x, y)
+  else:
+    raise NotImplementedError(
+        f"HLO comparison {direction} for opaque dtype {avals_in[0].dtype}")
```

Comment:
One potential issue with putting this here: the semantics of opaque dtype comparison might differ depending on the dtype. For example, we don't define `a < b` for `PRNGKeyArray` because the semantics of less-than are not well-defined for elements which are arrays of values. But for `bint`, it would be perfectly reasonable to define less-than.

Where do we encode that decision if the hlo lowering no longer delegates to `dtype._rules`?